### PR TITLE
Unit test, to see if mastodon can deal with windows / linux saved file paths (Issue 205)

### LINKED ISSUE #205: Windows - Linux - Compatibility
# User Story

Mastodon Users would like to be able to exchange project files between different operating systems. More specifically, Windows, Linux, MacOS.

# Acceptance

1. Project file can be saved from Linux environment and then opened from Windows environment.
1. Project file can be saved from Windows environment and then opened from Linux environment.

# Tasks

- [x] Test, if this actually works already.
- ~~Save path of SpimDataFile on project.xml in Linux-style, i.e. using forward slashes ("/")~~
grafik
- ~~Ensure that Linux-style paths are read correctly from Windows environments.~~
- [x] Write Unit test that allows checking the behaviour from different OS machines.

### PR DESCRIPTION
**Pull request by stefanhahmann**
Resolves #205

### CODE REVIEW

**Comment by stefanhahmann** [comment]
@maarzt may I ask you to run the 2 new tests on Linux (Ubuntu) machine.

**Comment by maarzt** [comment]
Yes, it works. But I don't like that the test relies on exceptions to be thrown. I need to show you how to create these BDV XML + HDF5 datasets. We could put a tiny dataset into the test/resources folder.

**Comment by maarzt** [comment]
@stefanhahmann here is a link on how to create the XML + HDF5 datasets

**Comment by stefanhahmann** [comment]
> Yes, it works. But I don't like that the test relies on exceptions to be thrown. I need to show you how to create these BDV XML + HDF5 datasets. We could put a tiny dataset into the test/resources folder.

Good idea. Would be a better test then.

File: src/main/java/org/mastodon/views/bdv/SharedBigDataViewerData.java [hunk 1]
```diff
@@ -370,6 +370,8 @@ public static SharedBigDataViewerData fromSpimDataXmlFile(
 			{
 				//trying to load the actual (not-dummy) data
 				spimData = new XmlIoSpimDataMinimal().load( spimDataXmlFilename );
+
+				System.out.println( "Successfully loaded spim data file: " + spimDataXmlFilename );
```
Comment (maarzt) [inline src/main/java/org/mastodon/views/bdv/SharedBigDataViewerData.java:374]: This ```system.out.println(...)``` should be removed

File: src/test/java/org/mastodon/views/trackscheme/bdv/SharedBigDataViewerDataTest.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import static org.junit.Assert.assertEquals;
+
+public class SharedBigDataViewerDataTest
+{
+	@Test
+	public void testFromSpimDataXmlFileWindowsStyle() throws SpimDataException, IOException
+	{
+		ViewerOptions viewerOptions = new ViewerOptions();
+		SharedBigDataViewerData sharedBigDataViewerData = SharedBigDataViewerData.fromSpimDataXmlFile( "./src/test/resources/org/mastodon/views/trackscheme/bdv/dataset_hdf5.xml", viewerOptions, null );
```
Comment (maarzt) [inline src/test/java/org/mastodon/views/trackscheme/bdv/SharedBigDataViewerDataTest.java:18]: In noticed, that there is already a BDV XML+H5 in the resources folder which we could use:
```src/test/resources/org/mastodon/mamut/importer/trackmate/mamutproject/FakeTracksBDV.xml```

**Review by maarzt — COMMENTED** [review]
I suggested two changes

File: src/main/java/org/mastodon/views/bdv/SharedBigDataViewerData.java (hunk 1 above)
Comment (stefanhahmann) [reply src/main/java/org/mastodon/views/bdv/SharedBigDataViewerData.java:374]: Removed.

File: src/test/java/org/mastodon/views/trackscheme/bdv/SharedBigDataViewerDataTest.java (hunk 2 above)
Comment (stefanhahmann) [reply src/test/java/org/mastodon/views/trackscheme/bdv/SharedBigDataViewerDataTest.java]: Thanks for the hint. Made changes to use these resources.

**Comment by maarzt** [comment]
Sorry this is now fixed in #210 and there is also a new unit test.

I close this PR in favor of #210

**Pull request closed without merging**